models/ensemble.py

```diff
@@ -51,16 +53,76 @@ def feature_importances(self):
 
     def predict(self, X):
+        """Predict the mean of the ensemble model predictions on inputs X"""
         model_preds = []
         for split, model in self.split2model.items():
             model_preds.append(model.predict(X))
 
         ensemble_pred = np.mean(model_preds, axis=0)
         return ensemble_pred
 
+    def predict_dist(self, X, distribution="laplace",
+                     uncertainty_scale=None):
+        """Predict a distribution parametrized by the mean and variance
+        of the ensemble model predictions on inputs X.
+
+        X (pandas DataFrame or numpy array): Inputs for prediction
+        distribution (str): Predicted distribution to parameterize.
+                            Options: [laplace, normal]
+        uncertainty_scale (float): Optional platt scale value to scale
+                                   uncertainties.
+        """
+        model_preds = []
+        for split, model in self.split2model.items():
+            model_preds.append(model.predict(X))
+
+        ensemble_mean = np.mean(model_preds, axis=0)
+        ensemble_var = np.var(model_preds, axis=0)
+
+        if distribution == "laplace":
+            Dist = Laplace
+            if uncertainty_scale is None:
+                ensemble_scale = np.sqrt(ensemble_var) / 2
+            else:
+                ensemble_scale = (
+                    np.sqrt(ensemble_var) / 2 * uncertainty_scale
+                )
+
+        elif distribution == "normal":
+            Dist = Normal
+            if uncertainty_scale is None:
+                ensemble_scale = np.sqrt(ensemble_var)
+            else:
+                ensemble_scale = np.sqrt(ensemble_var * uncertainty_scale)
+            ensemble_scale = np.log(ensemble_scale)
+
+        else:
+            raise ValueError(f"Distribution {distribution} not supported.")
+
+        ensemble_dist = Dist(
+            np.array([ensemble_mean, ensemble_scale])
+        )
+
+        return [dist for dist in ensemble_dist]
+
     def evaluate(self, X, y, metric):
         y_hat = self.predict(X)
         if metric not in metric_dict:
             raise ValueError(f"Metric {metric} not supported.")
         metric_fn = metric_dict[metric]
         return metric_fn(y, y_hat)
+
+    def uncertainty_scale(self, truth, pred_dist, distribution="laplace"):
+        """Compute MLE scaling factor (platt scaling)"""
+        mean = np.array([dist.mean() for dist in pred_dist])
+        scale = np.array([dist.scale for dist in pred_dist])
+        if distribution == "laplace":
+            numerator = np.absolute(truth.values - mean)
+            denominator = scale
+            return (numerator / denominator).mean()
+        elif distribution == "normal":
+            numerator = (truth.values - mean) ** 2
+            denominator = scale ** 2
+            return (numerator / denominator).mean()
+        else:
+            raise ValueError(f"Distribution {distribution} not supported.")
```

Comment:
Please double-confirm the implemented scaling of the distributions. It's leading to exceedingly large widths.mean() outputs.